### ISSUE TITLE
Change nm_join() and nm_data() to optionally  not return integer

### ISSUE THREAD

**Issue by kylebaron** [metrumresearchgroup/bbr#721]
Needs discussion; but I think it might simplify things downstream.

**Comment by seth127** [comment]
Thanks Kyle. Would making the change suggested in https://github.com/metrumresearchgroup/bbr/issues/722 accomplish this?

**Comment by kylebaron** [comment]
@seth127 No; I think #722 only addresses long integers, not integer to numeric / double. Related but not the same so I made two issues.

**Comment by kylebaron** [comment]
@seth127 one way this is related: if we ask data.table to return long integers as doubles, then it would make sense to return all ints as doubles out of consistency.
